### PR TITLE
[autograd] Explicitly error when layer_norm computes 3rd order derivatives

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.13.0) (oldest at bottom):
* #177806
* #177070
* #174079
* __->__ #176234

The saved mean and invstd tensors passed to layer_norm_double_backward
have no autograd history connecting them to the input (they come from
the forward kernel).  The double backward formula analytically accounts
for their dependency on input, so second-order gradients are correct.
However, when autograd differentiates *through* the double backward for
3rd+ order derivatives, it cannot see this dependency and treats
mean/invstd as constants, producing silently wrong results.

Fixes https://github.com/pytorch/pytorch/issues/175754

Authored with Claude.

### CODE REVIEW

**Review by albanD — APPROVED** [review]
SGTM
We could also make it differentiable. But this is strictly better than the current state so ok.

**Review by Skylion007 — APPROVED** [review]

File: torch/csrc/autograd/FunctionsManual.cpp [hunk 1]
```diff
@@ -4918,6 +4918,16 @@ std::tuple<Tensor, Tensor, Tensor> layer_norm_double_backward(
   auto gO = gO_t.reshape({M, N});
   auto save_mean = save_mean_t.reshape({M, 1});
   auto save_invstd = save_invstd_t.reshape({M, 1});
+  if (at::GradMode::is_enabled() && input.requires_grad()) {
+    // save_mean and save_invstd have no autograd history, error instead
+    // of silently giving wrong results for higher-order derivatives.
+    auto err = std::make_shared<DelayedError>(
+        "layer_norm does not support 3rd+ order derivatives.",
+        /* num inputs */ 3);
+    auto result = err->apply({save_mean, save_invstd, input});
```
Comment (Skylion007) [inline torch/csrc/autograd/FunctionsManual.cpp:4937]: Why error instead of fix it?

**Review by Skylion007 — APPROVED** [review]

File: torch/csrc/autograd/FunctionsManual.cpp (hunk 1 above)
Comment (soulitzer) [reply torch/csrc/autograd/FunctionsManual.cpp:4937]: having to recompute eps seemed annoying, but maybe its fine - going to land this one first since its a strict improvement
